SU2_CFD/src/solvers/CNSSolver.cpp

```diff
@@ -912,51 +910,63 @@ void CNSSolver::SetTauWall_WF(CGeometry *geometry, CSolver **solver_container, c
       su2double WallShearStress = GeometryToolbox::Norm(int(MAXNDIM), TauTangent);
 
       /*--- Calculate the quantities from boundary layer theory and
-       iteratively solve for a new wall shear stress. Use the current wall
-       shear stress as a starting guess for the wall function. ---*/
+       *    iteratively solve for a new wall shear stress. Use the current wall
+       *    shear stress as a starting guess for the wall function. ---*/
 
       unsigned long counter = 0;
-      su2double diff = 1.0, Eddy_Visc = 1.0;
+      su2double diff = 1.0;
       su2double U_Tau = sqrt(WallShearStress/Density_Wall);
-      su2double Y_Plus = 5.0;
+      su2double Y_Plus = 0.99*config->GetwallModelMinYPlus(); // to avoid warning
 
       su2double Y_Plus_Start = Density_Wall * U_Tau * WallDistMod / Lam_Visc_Wall;
 
-      /*--- Automatic switch off when y+ < 5 according to Nichols & Nelson (2004) ---*/
+      /*--- Automatic switch off when y+ < 5.0 according to Nichols & Nelson (2004) ---*/
 
-      if (Y_Plus_Start < Y_Plus) {
+      if (Y_Plus_Start < config->GetwallModelMinYPlus() ) {
         skipCounter++;
+        continue;
       }
       else while (fabs(diff) > tol) {
 
-        /*--- Friction velocity and u+ ---*/
-
         su2double U_Plus = VelTangMod/U_Tau;
 
         /*--- Gamma, Beta, Q, and Phi, defined by Nichols & Nelson (2004) page 1110 ---*/
 
         su2double Gam  = Recovery*U_Tau*U_Tau/(2.0*Cp*T_Wall);
-            /*--- nijso: heated wall needs validation testcase! ---*/
-        su2double Beta = q_w*Lam_Visc_Wall/(Density_Wall*T_Wall*Conductivity_Wall*U_Tau); // TODO: nonzero heatflux needs validation case
+        su2double Beta = q_w*Lam_Visc_Wall/(Density_Wall*T_Wall*Conductivity_Wall*U_Tau); 
         su2double Q    = sqrt(Beta*Beta + 4.0*Gam);
         su2double Phi  = asin(-1.0*Beta/Q);
 
+        /*--- Crocco-Busemann equation for wall temperature (eq. 11 of Nichols and Nelson) ---*/
+        /*--- update T_Wall due to aerodynamic heating, unless the wall is isothermal      ---*/
+
+        if (config->GetMarker_All_KindBC(iMarker) != ISOTHERMAL) {
+          su2double denum = (1.0 + Beta*U_Plus - Gam*U_Plus*U_Plus); 
+          if (abs(denum)>EPS){ 
```

Comment:
![No Rico, not today](https://tenor.com/view/madagscar-penguins-kaboom-gif-9833865.gif)

Reply:
yes you are right, it is not guaranteed that (beta*Uplus -gamma*Uplus^2) > -1   due to the presence of the heat flux. I will throw an SU2_MPI::Error

Reply:
might be better to warn rather than crash immediately, maybe it happens just in the first few iterations.
```suggestion
          if (denum > EPS){ 
```